### DISCUSSION TITLE
Using environment variables in config.yaml

### DISCUSSION THREAD

**Discussion by mbtools** [verdaccio/verdaccio#3555] [Ideas]
It would be nice to be able to reference environment variables for any configuration values in `config.yaml`. The main usage scenario is passing secrets to the configuration such as usernames and tokens. 

In some cases (like [here](https://verdaccio.org/docs/next/uplinks/#auth-property)), this is explicitly enabled. However, a general solution would be quite valuable and allow, for example, to pass secrets also to plugins (without them coding for it explicitly). Here's a [use case](https://github.com/deeplay-io/verdaccio-openid-connect/issues/13).

I imagine this working similarly to [GitHub actions](https://docs.github.com/en/actions/learn-github-actions/environment-variables) where values can be set using `parameter: $ENV_VARIABLE`.

**Comment by juanpicado** [comment]
ref: https://github.com/verdaccio/verdaccio/issues/1681

**Reply by Noriller** [reply]
Quick question: shouldn't it be reopened if the talk is still going?

**Reply by juanpicado** [reply]
Done

**Comment by mbtools** [comment]
Apparently, I didn't go back far enough when searching issues. Thank you for the ref!

**Reply by juanpicado** [reply]
Let's keep discussion there please feel free to copy the examples into that issue.
